src/ogdc_runner/common.py

```diff
@@ -0,0 +1,322 @@
+
+        with Steps(name="steps"):
+            check_dir_template()
+
+    # wait for the workflow to complete.
+    workflow_name = submit_workflow(workflow=w, wait=True)
+
+    # If overwrite is not True, we need to check the result of the
+    # `check-already-published` step to see if the data have been published or
+    # not.
+    # Check the result. Get an updated instance of the workflow, with the latest
+    # states for all notdes. Then, iterate through the nodes and find the
+    # template we define above ("check-already-published") and extract its
+    # output parameter.
+    completed_workflow = ARGO_WORKFLOW_SERVICE.get_workflow(name=workflow_name)
+    result = None
+    for node in completed_workflow.status.nodes.values():  # type: ignore[union-attr]
+        if node.template_name == "check-already-published":
+            result = node.outputs.parameters[0].value  # type: ignore[union-attr, index]
+    if not result:
+        err_msg = "Failed to check if data have been published"
+        raise OgdcWorkflowExecutionError(err_msg)
+
+    assert result in ("yes", "no")
+
+    # Cleanup this workflow, it is no longer needed
+    ARGO_WORKFLOW_SERVICE.delete_workflow(workflow_name)
+
+    return result == "yes"
+
+
+def data_already_published(
+    *,
+    recipe_config: RecipeConfig,
+    overwrite: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+) -> bool:
+    """Check for the existence of published data for the given
+    recipe and optionally remove it.
+
+    If `overwrite=True`, this function will remove any existing published data
+    for the provided recipe.
+
+    Returns `True` if data have already been published for the given recipe,
+    otherwise `False`.
+    """
+    if overwrite:
+        # If `overwrite` is True, remove the existing data and return `False`.
+        remove_existing_published_data(
+            recipe_config=recipe_config,
+            custom_image=custom_image,
+            custom_tag=custom_tag,
+        )
+        return False
+
+    return check_for_existing_published_data(
+        recipe_config=recipe_config,
+        custom_image=custom_image,
+        custom_tag=custom_tag,
+    )
+
+
+def parse_commands_from_recipe_file(recipe_dir: str, filename: str) -> list[str]:
+    """Read commands from a recipe file.
+
+    Args:
+        recipe_dir: The directory containing the recipe file
+        filename: The name of the recipe file to parse
+
+    Returns:
+        A list of commands from the recipe file, with comments removed
+    """
+    recipe_path = f"{recipe_dir}/{filename}"
+    logger.info(f"Reading recipe from {recipe_path}")
+
+    with fsspec.open(recipe_path, "rt") as f:
+        lines = f.read().split("\n")
+    commands = [line for line in lines if line and not line.startswith("#")]
+
+    return commands
+
+
+def apply_custom_container_config(
```

Comment:
Seems like this belongs in the `argo` module, alongside the functions it uses (`update_runner_image`, `update_namespace`).

Reply:
Removed the method, no longer needed